plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/recipes/UpgradeJenkinsTestHarnessVersion.java

```diff
@@ -0,0 +1,84 @@
+package io.jenkins.tools.pluginmodernizer.core.recipes;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.jenkins.UpgradeVersionProperty;
+import org.openrewrite.maven.AddProperty;
+import org.openrewrite.maven.MavenIsoVisitor;
+import org.openrewrite.maven.RemoveRedundantDependencyVersions;
+import org.openrewrite.xml.tree.Xml;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpgradeJenkinsTestHarnessVersion extends Recipe {
+    /**
+     * Logger
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(UpgradeJenkinsTestHarnessVersion.class);
+
+    /**
+     * The latest compatible version.
+     */
+    @Option(displayName = "Version", description = "The version.", example = "2.70")
+    String latestCompatibleVersion;
+
+    public UpgradeJenkinsTestHarnessVersion(String latestCompatibleVersion) {
+        this.latestCompatibleVersion = latestCompatibleVersion;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return "Upgrade Jenkins Test Harness version";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Upgrade Jenkins Test Harness version to ensure compatibility with jenkins.version.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenIsoVisitor<>() {
+
+            @Override
+            public Xml.Document visitDocument(Xml.Document document, ExecutionContext ctx) {
+
+                // add the property only if it doesn't exist
+                String propertyKey = "jenkins-test-harness.version";
+                String propertyValue = latestCompatibleVersion;
+
+                boolean propertyExists = document.getRoot().getChildren("properties").stream()
+                        .flatMap(properties -> properties.getChildren().stream())
+                        .anyMatch(property -> property.getName().equals(propertyKey));
+
+                if (!propertyExists) {
+                    LOG.info("Adding property: " + propertyKey + " with value: " + propertyValue);
+
+                    document = (Xml.Document) new AddProperty(propertyKey, propertyValue, null, null)
```

Comment:
Almost sure it must be removed. If the property is absent we don't touch it. Only removing existing dependency if they are added without using the property

Reply:
Definitely! 👍